src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,236 @@
+   */
+  private buildEndLink = (className: string|null, page: number) => {
+    const isNext = className === 'next';
+    const content = (
+      <Fragment>
+        <span class="vads-u-visibility--screen-reader">
+          {isNext ? 'Next' : 'Previous'}
+        </span>
+        {!isNext && <span aria-hidden="true">{'\u2039\u00a0\u00a0'}</span>}
+        <span class="hide-narrow">{isNext ? 'Next' : 'Prev'}</span>
+        {isNext && <span aria-hidden="true">{'\u00a0\u00a0\u203a'}</span>}
+      </Fragment>
+    );
+    return page ? (
+      <li><a {...this.getLinkProps({ className, page })}>{content}</a></li>
+    ) : (
+      <li aria-hidden="true"><span class={className}>{content}</span></li>
+    );
+  };
+
+  /**
+   * Build page links
+   */
+  private buildLink = (index: number, current: number, total: number) => {
+    let page = index + 1;
+    if (page < 1) {
+      page = 1;
+    } else if (page >= total) {
+      page = total;
+    }
+    const isCurr = page === current;
+    const props = this.getLinkProps({
+      page,
+      current: isCurr,
+      className: `va-pagination-link ${isCurr ? 'va-pagination-active' : ''}`,
+    });
+    return (
+      <li class={isCurr ? 'va-pagination-current' : null}>
+        <a {...props}>{page}</a>
+      </li>
+    );
+  };
+
+  /**
+   * Calculates and build visible pages
+   */
+  private buildPager = ({ current, total, edges, visible }) => {
+    let index : number;
+    const pages = [];
+
+    // max length - 2 edges - 2 ellipsis
+    const display = visible - (edges * 2) - 2;
+
+    const half = display / 2;
+    const start = Math.floor(
+      current > half
+        ? Math.max(Math.min(current - half, total - display), 0)
+        : 0
+    );
+    const end = Math.floor(
+      current > half
+        ? Math.min(current + half, total)
+        : Math.min(display, total)
+    );
+
+    // build start edge links & add ellipsis
+    if (start > 0 && edges > 0) {
+      for (index = 0; index < Math.min(edges, start); index++) {
+        pages.push(this.buildLink(index, current, total));
+      }
+      if (edges < start && (start - edges !== 1)) {
+        pages.push(this.addEllipsis());
+      } else if (start - edges === 1) {
+        pages.push(this.buildLink(edges, current, total));
+      }
+    }
+
+    // build middle links
+    for (index = start; index < end; index++) {
+      pages.push(this.buildLink(index, current, total));
+    }
+
+    // build ellipsis & ending edge links
+    if (end < total && edges > 0) {
+      if (
+        total - edges > end &&
+        (total - edges - end !== 1)
+      ) {
+        pages.push(this.addEllipsis());
+      } else if (total - edges - end === 1) {
+        pages.push(this.buildLink(end, current, total));
+      }
+      for (
+        index = Math.max(total - edges, end);
+        index < total;
+        index++
+      ) {
+        pages.push(this.buildLink(index, current, total));
+      }
+    }
+    return pages;
```

Comment:
The multiple nested conditionals and loops make this hard for me to follow. I'm personally a fan of this kind of pattern:

```jsx

myCollection.map(item => {
    <a>{item.whatever}</a>
});
```

Our React implementation does it like that...

https://github.com/department-of-veterans-affairs/component-library/blob/f3c3794991dfd82d20aa20fe4c7b1e9ca7de9a6f/src/components/Pagination/Pagination.jsx#L180-L196

_but_ it's backed up by something similar to what you have here:

https://github.com/department-of-veterans-affairs/component-library/blob/f3c3794991dfd82d20aa20fe4c7b1e9ca7de9a6f/src/components/Pagination/Pagination.jsx#L135-L161


I know that the markup has changed a bit, but do you think we could generate the pager portion with a `.map()` call?